File: front/src/app/shared/model/dto/cliente-dto.ts
import { UsuarioDto } from './usuario-dto';
import {StatusEnum} from "@/shared/model/enum/status.enum";

export interface ClienteDto {
    id?: number;
    nome?: string;
    email?: string;
    telefone?: string;
    usuario?: UsuarioDto;
    anotacoes?: string;
    status?: StatusEnum;
    criadoEm?: Date;
    alteradoEm?: Date;
    version?: number;
}

File: back/src/main/java/br/eng/eliseu/presente/model/dto/ClienteDto.java
package br.eng.eliseu.presente.model.dto;

import br.eng.eliseu.presente.model.StatusEnum;

import java.time.LocalDateTime;

public record ClienteDto(Long id,
                         String nome,
                         String email,
                         String telefone,
                         UsuarioDto usuario,
                         String anotacoes,
                         StatusEnum status,
                         LocalDateTime criadoEm,
                         LocalDateTime alteradoEm,
                         Long version          // Essencial para o Optimistic Locking do Hibernate
) {}

